pkg/data/generic.go

```diff
@@ -203,31 +196,92 @@ func (s *GenericSync) sync(resource dynamic.NamespaceableResourceInterface, quit
 }
 
 func (s *GenericSync) syncAdd(obj runtime.Object) error {
-	m, err := meta.Accessor(obj)
+	path, err := s.objPath(obj)
 	if err != nil {
 		return err
 	}
-	name := m.GetName()
-	var path = m.GetName()
-	if s.ns.Namespaced {
-		path = m.GetNamespace() + "/" + name
-	}
 	return s.opa.PutData(path, obj)
 }
 
 func (s *GenericSync) syncRemove(obj runtime.Object) error {
-	m, err := meta.Accessor(obj)
+	path, err := s.objPath(obj)
 	if err != nil {
 		return err
 	}
+	return s.opa.PatchData(path, opa_client.DataPatchOpRemove, nil)
+}
+
+func (s *GenericSync) syncAll(objs []unstructured.Unstructured) error {
+
+	// Build a list of patches to apply.
+	patches, err := s.generatePatches(objs)
+	if err != nil {
+		return err
+	}
+
+	// We don't do a 'remove' patch because doesn't exist we waste all the
+	// resources parsing the giant patch payload and have to try again.. Best
+	// option currently is to just do two requests.
+	err = s.opa.PutData("/", map[string]interface{}{})
```

Comment:
I was thinking about this a bit more and now I'm wondering why we can't just set the entire collection at once? If we do a PUT on /v1/data/kubernetes/<resource-type> it'll overwrite everything in the collection. This may even be more efficient than the PATCH method since the server and store don't have to process a bunch of separate patch operations. WDYT?

Reply:
I was also wondering about that. I shied-away from it initially because I was concerned about having to limit the size of the payloads and patches seemed like the most direct way to chunk them up. That being said.. there isn't really any difference between that and just building PUT payloads of different sizes. Let me update quickly to try it out and see if there is much of a difference in performance.

On paper I think you're right though that a single large PUT should be less load on the server as it is today.